### PR TITLE
fix: solve precision problem when decode bigint in json

### LINKED ISSUE #379: How to solve precision problem in tengo?
Hi,  thanks for such a great project!

How to solve precision problem in tengo?
```go
json := import("json")
fmt := import("fmt")


base_info := bytes(`{
        "1":7075984636689534001,
        "2":7075984636689534002
}`)
m := json.decode(base_info)
fmt.println(m[1])
```

result: 7075984636689534000

### PR DESCRIPTION
**Pull request by gufeijun**
Fixes #379 
slightly modify json.decode to solve precision problem when decode bigint in json

### CODE REVIEW

File: stdlib/json/decode.go [hunk 1]
```diff
@@ -62,9 +62,12 @@ func (d *decodeState) scanNext() {
 
 // scanWhile processes bytes in d.data[d.off:] until it
 // receives a scan code not equal to op.
-func (d *decodeState) scanWhile(op int) {
+func (d *decodeState) scanWhile(op int) (getFloat bool) {
```
Comment (geseq) [inline stdlib/json/decode.go:65]:
```suggestion
func (d *decodeState) scanWhile(op int) (isFloat bool) {
```

File: stdlib/json/decode.go [hunk 2]
```diff
@@ -62,9 +62,12 @@ func (d *decodeState) scanNext() {
...
 	s, data, i := &d.scan, d.data, d.off
 	for i < len(data) {
+		if data[i] == '.' {
+			getFloat = true
```
Comment (geseq) [inline stdlib/json/decode.go:69]:
```suggestion
			isFloat = true
```

File: stdlib/json/decode.go [hunk 3]
```diff
@@ -185,7 +189,7 @@ func (d *decodeState) object() (tengo.Object, error) {
 func (d *decodeState) literal() (tengo.Object, error) {
 	// All bytes inside literal return scanContinue op code.
 	start := d.readIndex()
-	d.scanWhile(scanContinue)
+	getFloat := d.scanWhile(scanContinue)
```
Comment (geseq) [inline stdlib/json/decode.go:192]:
```suggestion
	isFloat := d.scanWhile(scanContinue)
```

File: stdlib/json/decode.go [hunk 4]
```diff
@@ -210,8 +214,12 @@ func (d *decodeState) literal() (tengo.Object, error) {
 		if c != '-' && (c < '0' || c > '9') {
 			panic(phasePanicMsg)
 		}
-		n, _ := strconv.ParseFloat(string(item), 10)
-		return &tengo.Float{Value: n}, nil
+		if getFloat {
```
Comment (geseq) [inline stdlib/json/decode.go:217]:
```suggestion
		if isFloat {
```

**Review by geseq — COMMENTED** [review]
Small suggestion for clarity

**Comment by gufeijun** [comment]
Thanks for your review and suggestion

**Comment by geseq** [comment]
LGTM. just waiting for #407

**Review by geseq — DISMISSED** [review]

**Comment by geseq** [comment]
Can you rebase master please?

**Comment by gufeijun** [comment]
It's done.

**Comment by geseq** [comment]
Thanks! Can you look into the failing test please? You’ll likely need to change the tests to account for  int types

**Comment by gufeijun** [comment]
My bad, forgotten to test all modules :-)
The latest patch shall pass all tests.

**Review by geseq — APPROVED** [review]

**Comment by iyidan** [comment]
@geseq @gufeijun This fix seems introduced a new bug in json module. The detail is:

tengo code: 
`
fmt := import("fmt")
json := import("json")
fmt.println( json.decode("[1e3]"))  // expected: [1000]
`
but: 
in v2.16.0: output is [0]  (❌)
in v2.12.0: output is [1000] (✅ )

Could you please help fix it ?

**Comment by geseq** [comment]
https://github.com/d5/tengo/pull/424

**Pull request merged**